pkg/middleware/nodeid/nodeid.go

```diff
@@ -40,7 +42,14 @@ func FromContext(ctx context.Context) (string, error) {
 		if err != nil {
 			return "", err
 		}
-		defaultNodeID = spiceDBPrefix + hostname
+
+		// Hash the hostname to get the final default node ID.
+		hasher := xxhash.New()
```

Comment:
Nit: Move this out of the function?

Reply:
Since we cache the name globally after computation, I think this is fine